service/worker/scheduler/workflow.go

```diff
@@ -836,6 +840,14 @@ func (s *scheduler) processWatcherResult(id string, f workflow.Future, long bool
 		s.logger.Warn("just-closed workflow not found in running list", "workflow", id, "long", long)
 	}
 
+	// update workflow execution status in RecentActions
+	if s.hasMinVersion(ActionResultIncludesStatus) {
+		matchRecent := func(a *schedpb.ScheduleActionResult) bool { return a.StartWorkflowResult.WorkflowId == id }
```

Comment:
probably want
```suggestion
		matchRecent := func(a *schedpb.ScheduleActionResult) bool { return a.StartWorkflowResult.GetWorkflowId() == id }
```
for safety